### PR TITLE
Knative function controller resources

### PR DESCRIPTION
**Pull request by venturasr**
Create an Installer Module for Knative Function Controller

Related issue(s)
kyma-project/kyma#4803
#5075

Related Epic
kyma-project/console#804

### CODE REVIEW

**Comment by sayanh** [comment]
@venturasr Please note https://github.com/kyma-project/kyma/pull/4940 has an impact on this PR:

File: resources/knative-function-controller/README.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+## Prerequisites
+
+- Knative Build (v0.6.0)
+- Knative Serving (v0.6.1)
+- Istio (istio-1.0.7)
+
+## Installation
+
+### Install helm chart
```
Comment (bszwarc) [inline resources/knative-function-controller/README.md]:
```suggestion
### Install Helm chart
```

File: resources/knative-function-controller/README.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+export NAME=knative-function-controller
+export NAMESPACE=kyma-system
+helm install knative-function-controller \
+             --namespace="${NAMESPACE}" \
+             --name="${NAME}" \
+             --tls
+```
+
+
+### Create a ServiceAccount to enable knativeBuild docker builds
```
Comment (bszwarc) [inline resources/knative-function-controller/README.md]:
```suggestion
### Create a ServiceAccount to enable knativeBuild Docker builds
```

File: resources/knative-function-controller/README.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+Currently you cannot run builds in every existing namespace of your cluster. ServiceAccounts with linked docker
```
Comment (bszwarc) [inline resources/knative-function-controller/README.md]:
```suggestion
Currently, you cannot run builds for every Namespace existing in your cluster.
```

File: resources/knative-function-controller/README.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+repository credentials have to be created for each namespace that will be used for the knative-function controller.
```
Comment (bszwarc) [inline resources/knative-function-controller/README.md]:
```suggestion
Create ServiceAccounts with linked Docker repository credentials for each Namespace that will be used for the knative-function controller.
```

File: resources/knative-function-controller/README.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    name: knative-function-controller-docker-reg-credential
+    annotations:
+        build.knative.dev/docker-0: ${REGISTRY}
+data:
+    username: $(echo $username | base64)
+    password: $(echo $password | base64)
+EOF
+```
+
+## Your first function
```
Comment (bszwarc) [inline resources/knative-function-controller/README.md]:
```suggestion
## Running the first function
```

File: resources/knative-function-controller/README.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+Currently there is no UI support for the new knative-function-controller.
+please run your first function like this:
```
Comment (bszwarc) [inline resources/knative-function-controller/README.md]:
```suggestion
Run your first function in the following way:
```

File: resources/knative-function-controller/README.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+          return 'Hello World'
+        }
+      }
+  functionContentType: "plaintext"
+  size: "L"
+  runtime: "nodejs8"
+EOF
+``` 
+
+To get the url of your new service check the output of
```
Comment (bszwarc) [inline resources/knative-function-controller/README.md]:
```suggestion
To get the URL of your new service, run:
```

File: resources/knative-function-controller/README.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
+# knative-function-controller
+
+## Overview
+
+This project contains the chart for the Function Controller.
+
+>**NOTE**: This feature is experimental
```
Comment (bszwarc) [inline resources/knative-function-controller/README.md]:
```suggestion
>**NOTE**: This feature is experimental.
```

**Comment by k15r** [comment]
> * is it enabled by default? cause I don't think it should, it is alpha, code in incubator and kubeless is still a default right?

it is already maintained in the kyma-repository, but you are correct: it is alpha, and it is not enabled by default. Kubless is still the default `serverless-framework` ;)

**Comment by k15r** [comment]
> * why `functions.runtime.kyma-project.io` as the name, `runtime` is strange to me because in serverless you can have different runtimes, you can have a nodejs runtime or go or different. With this name it sounds to me like we will have different runtimes in a runtime. Or I completely do not understand the topic

I see your point, how about serverless.kyma-project.io as the api-group?

**Comment by derberg** [comment]
@k15r
- imho this component should be disabled by default and in release notes we should just say how to enable, we do it like this in this release with new service for asyncapi validation and conversion to not increase resource consumption without a reason
- `serverless.kyma-project.io` sounds perfect

File: resources/function-controller/templates/cluster_role.yaml [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+    verbs:
+      - get
+      - list
+      - watch
+      - create
+      - update
+      - patch
+      - delete
+  - apiGroups:
+      - runtime.kyma-project.io
```
Comment (Abd4llA) [inline resources/function-controller/templates/cluster_role.yaml]: api group changed to `serverless.kyma-project.io`.

File: resources/function-controller/templates/cluster_role.yaml [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
```
Comment (Abd4llA) [inline resources/function-controller/templates/cluster_role.yaml]: api group changed to `serverless.kyma-project.io`

File: resources/function-controller/README.md [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+
+## Running the first function
+
+Currently there is no UI support for the new function-controller.
+Run your first function in the following way:
+
+```bash
+export NAMESPACE=<NAMESPACE>
+cat <<EOF | kubectl apply -n ${NAMESPACE} -f -
+apiVersion: runtime.kyma-project.io/v1alpha1
```
Comment (Abd4llA) [inline resources/function-controller/README.md]: api group changed to `serverless.kyma-project.io`

**Comment by rakesh-garimella** [comment]
JSON schemas are handled here: https://github.com/kyma-project/kyma/issues/5461

File: resources/function-controller/README.md [hunk 12]
```diff
@@ -0,0 +1,60 @@
+# function-controller
```
Comment (bszwarc) [inline resources/function-controller/README.md]:
```suggestion
# Function Controller
```
We can use the regular name here, I checked the docs for other resources and they do it the same way.

File: resources/function-controller/README.md [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+
+## Prerequisites
+
+- Knative Build (v0.6.0)
+- Knative Serving (v0.6.1)
+- Istio (v1.0.7)
+
+## Installation
+
+### Install Helm chart
```
Comment (bszwarc) [inline resources/function-controller/README.md]:
```suggestion
### Install Helm chart
Follow the steps to install the chart:
```
Added an introductory line

File: resources/function-controller/README.md [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+
+Environment variables:
```
Comment (bszwarc) [inline resources/function-controller/README.md]:
```suggestion
1. Export the environment variables:
```

File: resources/function-controller/README.md [hunk 15]
```diff
@@ -0,0 +1,60 @@
...
+
+| Variable        | Description |
```
Comment (bszwarc) [inline resources/function-controller/README.md]:
```suggestion
| Variable        | Description | Sample value | 
```

File: resources/function-controller/README.md [hunk 16]
```diff
@@ -0,0 +1,60 @@
...
+| --------------- | ----------- |
```
Comment (bszwarc) [inline resources/function-controller/README.md]:
```suggestion
| --------------- | ----------- | --------|
```

File: resources/function-controller/README.md [hunk 17]
```diff
@@ -0,0 +1,60 @@
...
+| `FN_REGISTRY`   | URL of the container registry _Function_ images will be pushed to. Used for authentication. (e.g. `https://gcr.io/` for GCR, `https://index.docker.io/v1/` for Docker Hub) |
```
Comment (bszwarc) [inline resources/function-controller/README.md]:
```suggestion
| **FN_REGISTRY**   | The URL of the container registry Function images will be pushed to. Used for authentication.  | `https://gcr.io/` for GCR, `https://index.docker.io/v1/` for Docker Hub|
```

File: resources/function-controller/README.md [hunk 18]
```diff
@@ -0,0 +1,60 @@
...
+| `FN_REPOSITORY` | Name of the container repository _Function_ images will be pushed to. (e.g. `gcr.io/my-project` for GCR, `my-user` for Docker Hub) |
```
Comment (bszwarc) [inline resources/function-controller/README.md]:
```suggestion
| **FN_REPOSITORY** | The name of the container repository Function images will be pushed to. | `gcr.io/my-project` for GCR, `my-user` for Docker Hub |
```

File: resources/function-controller/README.md [hunk 19]
```diff
@@ -0,0 +1,60 @@
...
+
+1. Install knative-build
```
Comment (bszwarc) [inline resources/function-controller/README.md]:
```suggestion
2. Install Knative Build:
```

File: resources/function-controller/README.md [hunk 20]
```diff
@@ -0,0 +1,60 @@
...
+                 --namespace="knative-build" \
+                 --name="knative-build-init" \
+                 --tls
+    
+    helm install knative-build \
+                 --namespace="knative-build" \
+                 --name="knative-build" \
+                 --tls
+    ```
+2. Install the function controller charts
```
Comment (bszwarc) [inline resources/function-controller/README.md]:
```suggestion
3. Install the Function Controller charts:
```

File: resources/function-controller/README.md [hunk 21]
```diff
@@ -0,0 +1,60 @@
...
+    helm install function-controller \
+                 --namespace="${NAMESPACE}" \
+                 --name="${NAME}" \
+                 --set secret.registryAddress="${FN_REPOSITORY}" \
+                 --set secret.registryUserName="${reg_username}" \
+                 --set secret.registryPassword="${reg_password}" \
+                 --set config.dockerRegistry="${FN_REPOSITORY}" \
+                 --tls
+    ```
+## Running the first function
```
Comment (bszwarc) [inline resources/function-controller/README.md]:
```suggestion
## Run the first function
```

File: resources/function-controller/README.md [hunk 22]
```diff
@@ -0,0 +1,60 @@
...
+
+>**NOTE**: This feature is experimental.
+
+## Prerequisites
+
+- Knative Build (v0.6.0)
+- Knative Serving (v0.6.1)
+- Istio (v1.0.7)
+
+## Installation
```
Comment (bszwarc) [inline resources/function-controller/README.md]:
```suggestion
## Details
```
Changed to details to match the template

File: resources/function-controller/README.md [hunk 23]
```diff
@@ -0,0 +1,60 @@
...
+                 --name="${NAME}" \
+                 --set secret.registryAddress="${FN_REPOSITORY}" \
+                 --set secret.registryUserName="${reg_username}" \
+                 --set secret.registryPassword="${reg_password}" \
+                 --set config.dockerRegistry="${FN_REPOSITORY}" \
+                 --tls
+    ```
+## Running the first function
+
+Currently there is no UI support for the new function-controller.
```
Comment (bszwarc) [inline resources/function-controller/README.md]:
```suggestion
Currently, there is no UI support for the new Function Controller. To use it, follow these steps:
1. Execute steps 4-6 from [these](https://github.com/kyma-project/kyma/blob/master/components/function-controller/README.md) instructions.
2. Run your first function using [these](../../components/function-controller/README.md#create-a-sample-hello-world-function) steps.
```

File: resources/function-controller/README.md [hunk 24]
```diff
@@ -0,0 +1,60 @@
...
+Run your first function as mentioned [here](../../components/function-controller/README.md#create-a-sample-hello-world-function)
```
Comment (bszwarc) [inline resources/function-controller/README.md]:
```suggestion
```
moved up

File: resources/function-controller/README.md [hunk 25]
```diff
@@ -0,0 +1,62 @@
...
+    helm install function-controller \
+                 --namespace="${NAMESPACE}" \
+                 --name="${NAME}" \
+                 --set secret.registryAddress="${FN_REPOSITORY}" \
+                 --set secret.registryUserName="${reg_username}" \
+                 --set secret.registryPassword="${reg_password}" \
+                 --set config.dockerRegistry="${FN_REPOSITORY}" \
+                 --tls
+    ```
+## Run the first function
```
Comment (bszwarc) [inline resources/function-controller/README.md]:
```suggestion
### Run the first function
```

**Pull request merged**